src/bthread/task_group.cpp

```diff
@@ -1122,4 +1182,60 @@ void print_task(std::ostream& os, bthread_t tid) {
     }
 }
 
+bool TaskGroup::notify(bool force_wakeup) {
+    if (force_wakeup) {
+        _force_wakeup.store(true, std::memory_order_release);
+    }
+    if (!_waiting.load(std::memory_order_acquire)) {
+        return false;
+    }
+    // LOG(INFO) << "notifying waiting group: " << group_id_;
+    std::unique_lock<std::mutex> lk(_mux);
+    _cv.notify_one();
+    return true;
+}
+
+bool TaskGroup::NoTasks() {
+    return _remote_rq.empty() && _bound_rq.empty() && (has_tx_processor_work_ == nullptr || !has_tx_processor_work_());
+}
+
+bool TaskGroup::wait(){
+    _waiting.store(true, std::memory_order_release);
+    _waiting_workers.fetch_add(1, std::memory_order_release);
```

Comment:
``_waiting_workers`` is a counter. ``std::memory_order_relaxed`` is good enough for its purpose.